From be29683e776475c635affe33e01e211fca320428 Mon Sep 17 00:00:00 2001
From: Yuchou Gan <yuchou.gan@nxp.com>
Date: Fri, 11 May 2018 23:47:36 +0800
Subject: [PATCH 3820/5242] MLK-18283-2 gpu: imx: imx8_dprc: Align the dprc
 width to smaller size

commit  e9ae17866016c4c11c989648aa4af4a83c4693a1 from
https://source.codeaurora.org/external/imx/linux-imx.git

As to super-tiled format for blit engine, the dprc
width should aligned to smaller size instead of 64.

Signed-off-by: Yuchou Gan <yuchou.gan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/imx8_dprc.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/imx8_dprc.c b/drivers/gpu/imx/imx8_dprc.c
index 7136721..cf9be7e 100644
--- a/drivers/gpu/imx/imx8_dprc.c
+++ b/drivers/gpu/imx/imx8_dprc.c
@@ -445,7 +445,11 @@ void dprc_configure(struct dprc *dprc, unsigned int stream_id,
 			p1_w = round_up(dprc_width, info->cpp[0] == 2 ? 8 : 4);
 			break;
 		case DRM_FORMAT_MOD_VIVANTE_SUPER_TILED:
-			p1_w = round_up(dprc_width, 64);
+			if (dprc->is_blit_chan && dprc->devtype->has_fixup)
+				p1_w = round_up(dprc_width,
+						info->cpp[0] == 2 ? 8 : 4);
+			else
+				p1_w = round_up(dprc_width, 64);
 			break;
 		default:
 			p1_w = round_up(dprc_width,
-- 
1.7.9.5

